drivers/sensor/fdc2x1x/fdc2x1x.c

```diff
@@ -0,0 +1,1010 @@
+/* fdc2x1x.c - Driver for the Texas Instruments FDC2X1X */
+
+/*
+ * Copyright (c) 2020 arithmetics ITS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_fdc2x1x
+
+#include <device.h>
+#include <sys/util.h>
+#include <logging/log.h>
+
+#include "fdc2x1x.h"
+
+LOG_MODULE_REGISTER(FDC2X1X, CONFIG_SENSOR_LOG_LEVEL);
+
+static int fdc2x1x_init_config(const struct device *dev);
+
+/**
+ * Read/Write from device.
+ * @param dev - The device structure.
+ * @param reg - The register address. Use FDC2X1X_REG_READ(x) or
+ *				FDC2X1X_REG_WRITE(x).
+ * @param data - The register data.
+ * @param length - Number of bytes being read
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_bus_access(const struct device *dev, uint8_t reg,
+			      void *data, size_t length)
```

Comment:
Why not just declare this `uint8_t *data` and avoid casting it to `uint8_t*` below? The only two callers are allocating uint8_t arrays for this argument.

Reply:
declared as `uint8_t *data` now